### ISSUE TITLE
`use rustfmt;` or `use clippy;` does not trigger unresolved import in 2018 edition

### ISSUE THREAD

**Issue by topecongiro** [rust-lang/rust#58928]
Given the following test.rs,
```rust
use rustfmt;

fn main() {}
```

running `rustc --edition 2015 test.rs` gives the following output:

```
error[E0432]: unresolved import `rustfmt`
 --> test.rs:1:5
  |
1 | use rustfmt;
  |     ^^^^^^^ no `rustfmt` in the root

warning: unused import: `rustfmt`
 --> test.rs:1:5
  |
1 | use rustfmt;
  |     ^^^^^^^
  |
  = note: #[warn(unused_imports)] on by default

error: aborting due to previous error
```

however running `rustc --edition 2018 test.rs` does not trigger an unresolved import error:

```
warning: unused import: `rustfmt`
 --> test.rs:1:5
  |
1 | use rustfmt;
  |     ^^^^^^^
  |
  = note: #[warn(unused_imports)] on by default

error: aborting due to previous error
```

which seems strange to me. Tested with the following versions:

```
[~] % rustc +stable --version
rustc 1.33.0 (2aa4c46cf 2019-02-28)
[~] % rustc +nightly --version
rustc 1.35.0-nightly (87a436377 2019-03-03)
```

**Comment by topecongiro** [comment]
The same thing applies to `use clippy;` as well, so it seems to be related to the `tool_attributes` feature.

**Comment by petrochenkov** [comment]
>so it seems to be related to the tool_attributes feature.

Exactly, the imports refer to "tool modules".
The edition difference is due to uniform paths  - "uniform" means that `rustfmt` is resolved in the same way in `#[rustfmt::skip]` and in `use rustfmt;`.
So, this works as expected.

**Comment by petrochenkov** [comment]
You can also use a tool attribute in non-attribute context and see what happens:
```rust
fn main() {
    // The tool module also resolves uniformly in this context
    let x = rustfmt::skip; // ERROR expected value, found tool attribute `rustfmt::skip`
}
```

**Comment by Centril** [comment]
Closing per @petrochenkov's notes.
